### PR TITLE
Nicer exception when no scroll ID is provided

### PR DESCRIPTION
**Pull request by ghost**
#9469

### CODE REVIEW

File: src/main/java/org/elasticsearch/action/search/type/TransportSearchHelper.java [hunk 1]
```diff
@@ -62,7 +63,7 @@ public static String buildScrollId(SearchType searchType, AtomicArray<? extends
         } else if (searchType == SearchType.SCAN) {
             return buildScrollId(ParsedScrollId.SCAN, searchPhaseResults, attributes);
         } else {
-            throw new ElasticsearchIllegalStateException("search_type [" + searchType + "] not supported");
+            throw new ElasticsearchIllegalStateException();
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/action/search/type/TransportSearchHelper.java]: I don't think we should have an empty exception message?

File: src/main/java/org/elasticsearch/action/search/type/TransportSearchHelper.java [hunk 2]
```diff
@@ -94,15 +95,16 @@ public static ParsedScrollId parseScrollId(String scrollId) {
             throw new ElasticsearchIllegalArgumentException("Failed to decode scrollId", e);
         }
         String[] elements = Strings.splitStringToArray(spare.get(), ';');
+        System.out.println(Arrays.toString(elements));
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/action/search/type/TransportSearchHelper.java]: Leftover s.o.p

File: src/main/java/org/elasticsearch/action/search/type/TransportSearchHelper.java [hunk 3]
```diff
@@ -133,4 +135,4 @@ private TransportSearchHelper() {
 
     }
 
-}
+}
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/action/search/type/TransportSearchHelper.java]: removed newline

**Comment by rjernst** [comment]
@Myll I left some comments.

**Comment by Myll** [comment]
@rjernst okay, fixed

File: src/main/java/org/elasticsearch/action/search/type/TransportSearchHelper.java [hunk 4]
```diff
@@ -39,6 +39,7 @@
 import org.elasticsearch.search.internal.ShardSearchTransportRequest;
 
 import java.io.IOException;
+import java.util.Arrays;
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/action/search/type/TransportSearchHelper.java]: leftover import

**Comment by rjernst** [comment]
@Myll Can you remove the leftover import and rebase?

**Comment by clintongormley** [comment]
Hi @stephenfournier 

Any chance you want to pick this up again, or should we take it over?

**Comment by clintongormley** [comment]
No further feedback. Closing

**Pull request closed without merging**